pkg/virt-controller/services/multus_annotations.go

```diff
@@ -58,10 +58,13 @@ func (mnap multusNetworkAnnotationPool) toString() (string, error) {
 func generateMultusCNIAnnotation(vmi *v1.VirtualMachineInstance) (string, error) {
 	multusNetworkAnnotationPool := multusNetworkAnnotationPool{}
 
-	multusNonDefaultNetworks := vmispec.FilterMultusNonDefaultNetworks(vmi.Spec.Networks)
-	for i, network := range multusNonDefaultNetworks {
-		multusNetworkAnnotationPool.add(
-			newMultusAnnotationData(vmi, network, fmt.Sprintf("net%d", i+1)))
+	netowrkNameScheme := namescheme.CreateNetworkNameScheme(vmi)
+	for _, network := range vmi.Spec.Networks {
+		podInterfaceName := netowrkNameScheme[network.Name]
+		if podInterfaceName != namescheme.PrimaryPodInterfaceName {
```

Comment:
In the scenario where there is a VMI with a secondary network (of type Multus) configured as the default (default=true) network,
and have the exact same name as `namescheme.PrimaryPodInterfaceName` (eth0),
does it make sense to filter it out?

Reply:
> In the scenario where there is a VMI with a secondary network (of type Multus) configured as the default (default=true) network

There is not such thing. When it is marked "default", it means it is the primary network.

> and have the exact same name as `namescheme.PrimaryPodInterfaceName` (eth0), does it make sense to filter it out?

If both are defined (pod and multus-default), then it is a configuration error.

Reply:
What I meant is a VMI with Multus network and its name is eth0, at first, I thought it will filter it out 
but I got it wrong. Sorry for the noise.